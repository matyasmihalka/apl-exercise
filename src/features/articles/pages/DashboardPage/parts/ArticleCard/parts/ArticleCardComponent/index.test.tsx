import renderer from 'react-test-renderer'

import doggo from '~/features/articles/fixtures/assets/doggo.webp'
import type { ArticleDetailTye } from '~/features/articles/types'

import { ArticleCardComponent } from '.'

const article = {
  articleId: '15',
  createdAt: '2022-07-15T12:50:25.23354',
  imageId: '15',
  lastUpdatedAt: '2022-07-15T12:50:25.23354',
  perex:
    'Clever, lively, courageous, entertaining: that’s the Dachshund. The Dachshund’s name comes from the German words for badger (dachs) and dog (hund)',
  title: 'Dachshund Dog Breed Facts & Information',
}
// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
const ArticleDetail = { comments: [] } as ArticleDetailTye

describe('[UI] Article card component', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <ArticleCardComponent
          article={article}
          imgURL={doggo}
          articleDetail={ArticleDetail}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
