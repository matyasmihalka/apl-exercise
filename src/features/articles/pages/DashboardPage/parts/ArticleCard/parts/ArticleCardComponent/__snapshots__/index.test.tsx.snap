// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UI] Article card component should match snapshot 1`] = `
<article
  className="sc-abd3bd29-0 hlZyiZ"
>
  <div
    className="sc-abd3bd29-2 eolfgv"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": 244,
          "margin": 0,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": 272,
        }
      }
    >
      <img
        alt="cute dog"
        data-nimg="fixed"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "cover",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </span>
  </div>
  <section
    className="sc-abd3bd29-1 eyVhfC"
  >
    <h2
      className="sc-abd3bd29-3 GMVPF"
    >
      Dachshund Dog Breed Facts & Information
    </h2>
    <div
      className="sc-a61f1b2d-0 gLYASV sc-abd3bd29-4 dQYpdj"
    >
       
      John Newmann 
      <span>
         •
      </span>
       
      15/07/2022
    </div>
    <p
      className="sc-abd3bd29-5 kYYlrn"
    >
      Clever, lively, courageous, entertaining: that’s the Dachshund. The Dachshund’s name comes from the German words for badger (dachs) and dog (hund)
    </p>
    <div>
      <a
        className="sc-abd3bd29-6 ljTNhW"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Read whole article
      </a>
      <span
        className="sc-abd3bd29-7 dvysSG"
      >
        0
         
        Comments
      </span>
    </div>
  </section>
</article>
`;
